pennylane/workflow/_capture_qnode.py

```diff
@@ -97,26 +97,26 @@ def _get_qnode_prim():
     qnode_prim = jax.core.Primitive("qnode")
     qnode_prim.multiple_results = True
 
-    # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-arguments, unused-argument
     @qnode_prim.def_impl
     def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts, batch_dims=None):
+        if shots != device.shots:
+            raise NotImplementedError(
+                "override shots are not yet supported with the program capture execution."
+            )
+        if qnode_kwargs["diff_method"] not in {"backprop", "best"}:
+            raise NotImplementedError(
+                "only backpropagation derivatives are supported at this time."
+            )
```

Comment:
Probably this is way too messy, but we could fall back to the current execution path if we encounter differing shots or diff_methods, no?

Reply:
I'd like to have a clearer signal of "this isn't supported" for the moment.